documentation/index.rst

```diff
@@ -0,0 +1,158 @@
+
+1. User documentation that explains how people will use Xapian,
+   typically either as a "full worked example", or as a deep dive into
+   a particular feature.
+
+   Most of this documentation lives in the*Xapian user guide*, which
+   is available `online`_ (built with python examples)
+   and as `a git repo on Github`_ which
+   you can build for a range of different languages.
+   
+2. API documentation, which is built from specially-formatted comments
+   in Xapian's C++ header files. This allows people writing code that
+   uses Xapian to check on details of the classes and methods they are
+   using, as well as to explore what features there are in Xapian.
+
+3. Developer documentation, for people who want to
+   contribute directly to Xapian.
+   `This guide`_ is part of that
+   (and is also `available on github`_).
+
+   There is also some developer documentation available elsewhere,
+   particularly in the main Xapian source repository -- look in
+   particular in ``xapian-core/docs``. Some of this is being migrated
+   to the developer guide. (Some is also being migrated to the user
+   guide.)
+
+4. `The Xapian wiki`_ also contains information, including links
+   to articles and talks that people have written about using Xapian
+   in various ways. Although this isn't always official Xapian
+   documentation, it may be helpfuil. Some of it may now be quite out
+   of date, so if something you find there doesn't work, you may want
+   to check the official documentation to see if you can figure out
+   how to change the advice or code you find to work with a recent
+   version of Xapian.
+
+5. Omega, a standalone indexing and web search tool built using
+   Xapian, has its own documentation bundled with its source code,
+   and `available on our website`_.
+
+6. The language bindings,
+   which allow you to use Xapian from programming languages others than C++,
+   have `their own documentation`_
+   which mostly focusses on ways that
+   the API is different from C++ in those languages.
+   
+.. _online: https://getting-started-with-xapian.readthedocs.org/
+.. _a git repo on Github: https://github.com/xapian/xapian-docsprint/
+.. _This guide: https://xapian-developer-guide.readthedocs.org/
+.. _available on github: https://github.com/xapian/xapian-developer-guide/
+.. _The Xapian wiki: https://trac.xapian.org/wiki
+.. _available on our website: https://xapian.org/docs/omega/
+.. _their own documentation: https://xapian.org/docs/bindings/
+
+
+Building the User guide and developer guide
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The user guide and developers guide can both be built using Sphinx.
+Generally, you will just need to:
+
+.. code-block:: bash
+
+   pip install -r requirements.txt
+   make html
+
+For the user guide, you can control which language the examples use by
+setting the ``LANGUAGE`` variable when you call make:
+
+.. code-block:: bash
+
+   make html LANGUAGE=c++
```

Comment:
I wonder if we want to start duplicating documentation about these sort of details?  If we just said something like "see `make help` for details of the options available" that would avoid having an extra place to update if we change the details of how the language is selected here.

Reply:
I've removed the duplication; in the process I noticed that neither user nor dev guides had amazing documentation. I've fixed the dev guide in another commit in this branch, and I'll open a separate PR for the user guide that makes the two more consistent.